l1-contracts/deploy-scripts/PrepareZKChainRegistrationCalldata.s.sol

```diff
@@ -0,0 +1,286 @@
+        config.l1SharedBridgeProxy = toml.readAddress("$.deployed_addresses.l1_shared_bridge_proxy_addr");
+        config.erc20BridgeProxy = toml.readAddress("$.deployed_addresses.erc20_bridge_proxy_addr");
+
+        config.chainId = toml.readUint("$.chain.chain_id");
+        config.eraChainId = toml.readUint("$.chain.era_chain_id");
+        config.chainAdmin = toml.readAddress("$.chain.admin");
+        config.diamondCutData = toml.readBytes("$.chain.diamond_cut_data");
+        config.bridgehubCreateNewChainSalt = toml.readUint("$.chain.bridgehub_create_new_chain_salt");
+        config.baseToken = toml.readAddress("$.chain.base_token_addr");
+        config.governance = toml.readAddress("$.chain.governance_addr");
+
+        bytecodes.l2SharedBridgeBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/contracts/bridge/L2SharedBridge.sol/L2SharedBridge.json"
+        );
+
+        bytecodes.l2SharedBridgeProxyBytecode = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol/TransparentUpgradeableProxy.json"
+        );
+        bytecodes.beaconProxy = Utils.readHardhatBytecode(
+            "/../l2-contracts/artifacts-zk/@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol/BeaconProxy.json"
+        );
+    }
+
+    function checkBaseTokenAddress() internal view {
+        if (config.baseToken == address(0)) {
+            revert("Token address is not set");
+        }
+
+        // Check if it's ethereum address
+        if (config.baseToken == ADDRESS_ONE) {
+            return;
+        }
+
+        if (config.baseToken.code.length == 0) {
+            revert("Token address is an EOA");
+        }
+
+        console.log("Using base token address:", config.baseToken);
+    }
+
+    function prepareRegisterBaseTokenCall() internal view returns (IGovernance.Call memory) {
+        Bridgehub bridgehub = Bridgehub(config.bridgehub);
+
+        bytes memory data = abi.encodeCall(bridgehub.addToken, (config.baseToken));
+
+        return IGovernance.Call({target: config.bridgehub, value: 0, data: data});
+    }
+
+    function computeL2BridgeAddress() internal view returns (address) {
+        bytes32 salt = "";
+        bytes32 bridgeBytecodeHash = L2ContractHelper.hashL2Bytecode(bytecodes.l2SharedBridgeBytecode);
+        bytes memory bridgeConstructorData = abi.encode(config.eraChainId);
+
+        address deployer;
+        address l2GovernorAddress;
+
+        if (isEOA(msg.sender)) {
+            deployer = msg.sender;
+        } else {
+            deployer = AddressAliasHelper.applyL1ToL2Alias(msg.sender);
+        }
+
+        if (isEOA(config.governance)) {
+            l2GovernorAddress = config.governance;
+        } else {
+            l2GovernorAddress = AddressAliasHelper.applyL1ToL2Alias(config.governance);
```

Comment:
add a comment on why we're using aliasing in this case.
(as in theory, we could have used aliasing always, right? - to force governance to go via L1->L2 messages)